src/D3MOracle.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-FileCopyrightText: © 2021-2022 Dai Foundation <www.daifoundation.org>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021-2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity ^0.8.14;
+
+interface VatLike {
+    function live() external view returns (uint256);
+}
+
+interface HubLike {
+    function culled(bytes32) external view returns (uint256);
+}
+
+contract D3MOracle {
+    // --- Auth ---
+    /**
+        @notice Maps address that have permission in the Pool.
+        @dev 1 = allowed, 0 = no permission
+        @return authorization 1 or 0
+    */
+    mapping (address => uint256) public wards;
+    address public hub;
```

Comment:
Can define this as `HubLike` (as we are doing that for `vat`).

Reply:
Not really opposed to do it but here my reasoning on why I left this way:
- vat is immutable and used twice, both times used calling a method.
- hub is used once and I liked to keep `file` without castings.

Anyway if you feel looks better I'll proceed.

Reply:
Yeah I fill the uniformity in the variable definiton looks a bit better and casting in `file` is something we do a lot.
So in favor on changing, but not critical at all.

Reply:
Ok good, I've proceed giving uniformity but changing the `vat` to an `address`. Hope that looks good as well.